packages/mui-material/src/styles/createThemeNoVars.js

```diff
@@ -24,10 +24,13 @@ function createThemeNoVars(options = {}, ...args) {
     ...other
   } = options;
 
-  if (options.vars) {
+  if (
+    options.vars &&
+    options.generateThemeVars === undefined // check for `generateThemeVars` (created internally) to handle nested theme
```

Comment:
```suggestion
    // The error should throw only for the root theme creation because user is not allowed to use a custom node `vars`.
    // `generateThemeVars` is the closest identifier for checking that the `options` is a result of `createTheme` with CSS variables.
    options.generateThemeVars === undefined
```

Reply:
@aarongarciah is it clearer?

Reply:
Much better! Thanks